src/System.Management.Automation/engine/parser/SemanticChecks.cs

```diff
@@ -214,15 +241,19 @@ public override AstVisitAction VisitAttribute(AttributeAst attributeAst)
                 string name = namedArg.ArgumentName;
                 if (names.Contains(name))
                 {
-                    _parser.ReportError(namedArg.Extent, () => ParserStrings.DuplicateNamedArgument, name);
+                    _parser.ReportError(namedArg.Extent,
+                        nameof(ParserStrings.DuplicateNamedArgument),
+                        ParserStrings.DuplicateNamedArgument,
+                        name);
                 }
                 else
                 {
                     names.Add(name);
 
                     if (!namedArg.ExpressionOmitted && !IsValidAttributeArgument(namedArg.Argument, constantValueVisitor))
                     {
-                        _parser.ReportError(namedArg.Argument.Extent, GetNonConstantAttributeArgErrorExpr(constantValueVisitor));
+                        Tuple<string, string> errorInfo = GetNonConstantAttributeArgErrorExpr(constantValueVisitor);
+                        _parser.ReportError(namedArg.Argument.Extent, errorInfo.Item1, errorInfo.Item2);
```

Comment:
`errorInfo.Item1, errorInfo.Item2` looks bad - maybe use out parameters in `GetNonConstantAttributeArgErrorExpr`?